lib/Sema/TypeCheckAttr.cpp

```diff
@@ -1377,8 +1377,11 @@ void AttributeChecker::visitObjCAttr(ObjCAttr *attr) {
     if (!checkObjCDeclContext(D))
       error = diag::invalid_objc_decl_context;
     else if (auto accessor = dyn_cast<AccessorDecl>(func))
-      if (!accessor->isGetterOrSetter())
-        error = diag::objc_observing_accessor;
+      if (!accessor->isGetterOrSetter()) {
+        diagnoseAndRemoveAttr(attr, diag::objc_observing_accessor, accessor->getDescriptiveKind()).limitBehavior(behavior);
```

Comment:
Nit: this needs formatting.

Reply:
Done here.
https://github.com/swiftlang/swift/pull/74462/commits/101c8074d0f1107766d7a50d7ffc4b74640ebdba